UI/js/controllers/controllers.js

```diff
@@ -101,13 +100,18 @@ dashboardApp.controller('scheduleController', ['Schedule', '$scope', '$rootScope
     }];
 
    $scope.$on('canvasCourses', function (event, canvasCourses) {
+      //listen for changes to the Canvas course array (created by the Courses controller)
+      // and match the course title of the Canvas assignments to the course title in the array
       $.each($scope.schedule, function() {
         if(this.contextLMS === 'canvas'){
           var thisId = _.last(this.contextUrl.split('/'));
           var thisContext = _.findWhere(canvasCourses, {id: thisId});
```

Comment:
It is an array that is created by the Courses controller. Which may finish its work before or after the  Schedule controller. $scope.$on('canvasCourses', <callback>)  is a watch for that array coming into existence and then  fires the callback  that modifies the data bound to the Schedule view - updating Canvas course titles.